src/Compilers/CSharp/Portable/CSharpCompilationOptions.cs

```diff
@@ -187,6 +149,7 @@ public sealed class CSharpCompilationOptions : CompilationOptions, IEquatable<CS
             generalDiagnosticOption: other.GeneralDiagnosticOption,
             warningLevel: other.WarningLevel,
             specificDiagnosticOptions: other.SpecificDiagnosticOptions,
+            perTreeDiagnosticOptions: other.PerTreeDiagnosticOptions.Select(kvp => (kvp.Key, kvp.Value.Select(options => (options.Key, options.Value)))),
```

Comment:
this seems weird.  everyone else just passes this stuff along.  you do computation on it.